lib/spack/spack/solver/asp.py

```diff
@@ -1315,7 +1299,9 @@ class Body(object):
         # compiler flags
         for flag_type, flags in spec.compiler_flags.items():
             for flag in flags:
-                clauses.append(f.node_flag(spec.name, flag_type, flag))
+                clauses.append(f.node_flag(spec.name, flag_type, flag[0]))
+                if flag[1] is True:
+                    clauses.append(f.node_flag_propagate(spec.name, flag_type))
```

Comment:
We shouldn't regenerate this for each individual flag
```suggestion
            if flags and flags[0][1] is True:
                clauses.append(f.node_flag_propagate(spec.name, flag_type))
```